x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/api_key/30_update.yml

```diff
@@ -346,3 +346,508 @@ teardown:
+                    "privileges": ["read"]
+                  }
+                ]
+              }
+            },
+            "metadata": {
+               "letter": "a",
+               "number": 42
+            }
+          }
+  - length: { "noops": 0 }
+  - length: { "updated": 2 }
+  - match: {
+    "updated.0": "$key_id_1"
+  }
+  - match: {
+    "updated.1": "$key_id_2"
+  }
+
+  # Bulk update without request fields does not update API keys
+  - do:
+      headers:
+        Authorization: "Basic YXBpX2tleV91c2VyXzE6eC1wYWNrLXRlc3QtcGFzc3dvcmQ=" # api_key_user_1
+      security.bulk_update_api_keys:
+        body: >
+          {
+            "ids": ["$key_id_1", "$key_id_2"]
+          }
+  - length: { "updated": 0 }
+  - length: { "noops": 2 }
+  - match: {
+    "noops.0": "$key_id_1"
+  }
+  - match: {
+    "noops.1": "$key_id_2"
+  }
+
+  # Check updated privileges
+  - do:
+      headers:
+        Authorization: ApiKey ${login_creds_1}
+      security.has_privileges:
+        user: null
+        body: >
+          {
+            "cluster": ["manage_own_api_key"],
+            "index": [
+              {
+                "names": ["index-a"],
+                "privileges": ["write"]
+              },
+              {
+                "names": ["index-b"],
+                "privileges": ["read"]
+              }
+            ]
+          }
+  - match: { "has_all_requested": true }
+
+  - do:
+      headers:
+        Authorization: ApiKey ${login_creds_2}
+      security.has_privileges:
+        user: null
+        body: >
+          {
+            "cluster": ["manage_own_api_key"],
+            "index": [
+              {
+                "names": ["index-a"],
+                "privileges": ["write"]
+              },
+              {
+                "names": ["index-b"],
+                "privileges": ["read"]
+              }
+            ]
+          }
+  - match: { "has_all_requested": true }
+
+  # Check that metadata was updated
+  - do:
+      headers:
+        Authorization: "Basic YXBpX2tleV91c2VyXzE6eC1wYWNrLXRlc3QtcGFzc3dvcmQ=" # api_key_user_1
+      security.get_api_key:
+        owner: true
+  - length: { "api_keys": 2 }
+  - match: {
+    "api_keys.0.metadata": {
+      "letter": "a",
+      "number": 42
+    }
+  }
+  - match: {
+    "api_keys.1.metadata": {
+      "letter": "a",
+      "number": 42
+    }
+  }
+---
+"Test bulk update api key without explicit field updates":
+
+  - do:
+      headers:
+        Authorization: "Basic YXBpX2tleV91c2VyXzE6eC1wYWNrLXRlc3QtcGFzc3dvcmQ=" # api_key_user_1
+      security.create_api_key:
+        body: >
+          {
+            "name": "api-key-1",
+            "role_descriptors": {
+              "role-a": {
+                "cluster": ["all"],
+                "index": [
+                  {
+                    "names": ["index-a"],
+                    "privileges": ["read"]
+                  }
+                ]
+              }
+            },
```

Comment:
I slightly prefer to have empty role_descriptors when creating the API keys. This test case is for auto-update limited_by_role_descriptors. Skipping the key's own descriptors here ensure everything works off the owner's permissions (and update).

Reply:
Part of the intention for this test is to show that the existing `role_descriptors` field doesn't get overwritten. I can change creation to not assign descriptors but the test in its current state captures a pretty common scenario: given an API key with previously assigned roles, and an update to its owner's roles, bring it up to date with the owner's roles _without_ touching the assigned descriptors. I could make this intention more clear via a comment or in the test name. Or I could drop role descriptors, I don't feel strongly here. WDYT?

Reply:
Hmm on second thought, since we don't really assert on the role descriptor values themselves (only on the effective privs of the API key), the test isn't really showing that they don't change. So I'm fully on board with your suggestion 👍